src/doc/doc-printer.js

```diff
@@ -244,6 +266,10 @@ function printDocToString(doc, options) {
         case "align":
           cmds.push([makeAlign(ind, doc.n, options), mode, doc.contents]);
 
+          break;
+        case "trim":
+          trim(out);
```

Comment:
I believe we'll have to update the `pos` (in `printDocToString`) and the `width` (in `fits`) so that the line length can be calculated correctly, we didn't need to do so previously since it's at the end of the line but it's not the case now.

And we should add some tests for it, though it seems we didn't have tests for it from the beginning.

I'm not too familiar with the algorithm here so I might miss something.

Reply:
I've rebased the commit and fixed the trivial error.

I've trying to find a way to test this new `trim` command. All the tests appear to be integration tests (run Prettier on a file and compare snapshot), not unit tests (typical assertion-based unit tests), which means I can't really test the command because it's not used by any of the official formatters (only prettier-csharp). Any idea?

Reply:
You could add a test file for `doc` in [`tests_integration/__tests__`](https://github.com/prettier/prettier/tree/master/tests_integration/__tests__) (see [`tests_integration/__tests__/load-toml.js`](https://github.com/prettier/prettier/blob/master/tests_integration/__tests__/load-toml.js) for example), though it's not ideal but we could polish our tests later in other PRs.